File: online-food-ordering-server/src/main/java/com/ofos/service/NotificationService.java
package com.ofos.service;

import com.ofos.dto.request.BulkNotificationRequest;
import com.ofos.dto.request.SendNotificationRequest;
import com.ofos.dto.request.UpdatePreferenceRequest;
import com.ofos.dto.response.NotificationPreferenceResponse;
import com.ofos.dto.response.NotificationResponse;
import com.ofos.entity.NotificationType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    
    // Send Notifications
    NotificationResponse sendNotification(SendNotificationRequest request);

    NotificationResponse createInAppNotification(Long userId, String title, String message, NotificationType type, Long orderId);
    
    void sendOrderConfirmation(Long orderId);
    
    void sendOrderStatusUpdate(Long orderId, String oldStatus, String newStatus);
    
    void sendOrderDelivered(Long orderId);
    
    void sendOrderCancelled(Long orderId, String reason);
    
    void sendPaymentSuccess(Long orderId);
    
    void sendPaymentFailed(Long orderId, String reason);
    
    void sendWelcomeEmail(String userEmail, String userName);
    
    void sendOtpEmail(String email, String otp);
    
    void sendOtpSms(String phoneNumber, String otp);
    
    // Bulk Notifications
    void sendBulkNotification(BulkNotificationRequest request);
    
    // User Notifications
    Page<NotificationResponse> getUserNotifications(String userEmail, Pageable pageable);
    
    NotificationResponse getNotificationById(Long notificationId);
    
    void markAsRead(Long notificationId, String userEmail);
    
    void markAllAsRead(String userEmail);

    void deleteNotification(Long notificationId, String userEmail);
    
    Long getUnreadCount(String userEmail);
    
    // Preferences
    NotificationPreferenceResponse getPreferences(String userEmail);
    
    NotificationPreferenceResponse updatePreferences(UpdatePreferenceRequest request, String userEmail);
    
    // System
    void processPendingNotifications();
    
    void deleteOldNotifications(Integer daysOld);
}

File: ofos-client/src/features/notification/services/notificationService.js
import api from '../../../api/axiosConfig';

export const notificationService = {
  // Get user notifications
  getNotifications: (page = 0, size = 20) => {
    return api.get('/notifications', { params: { page, size } });
  },

  // Get unread count
  getUnreadCount: () => {
    return api.get('/notifications/unread/count');
  },

  // Mark single notification as read
  markAsRead: (notificationId) => {
    return api.patch(`/notifications/${notificationId}/read`);
  },

  // Mark all notifications as read
  markAllAsRead: () => {
    return api.patch('/notifications/read-all');
  },

  // Get notification preferences
  getPreferences: () => {
    return api.get('/notifications/preferences');
  },

  // Update notification preferences
  updatePreferences: (preferences) => {
    return api.put('/notifications/preferences', preferences);
  },

  // Delete notification
  deleteNotification: (notificationId) => {
    return api.delete(`/notifications/${notificationId}`);
  },
};